packages/build/src/buildbot_client/main.js

```diff
@@ -0,0 +1,32 @@
+const net = require('net')
+const { promisify } = require('util')
+
+const pEvent = require('p-event')
+
+const { addErrorInfo } = require('../error/info')
+
+const startBuildbotClient = async function(buildbotServerSocket) {
+  if (buildbotServerSocket === undefined) {
+    return
+  }
+  try {
+    const buildbotClient = net.createConnection(buildbotServerSocket)
+    await pEvent(buildbotClient, 'connect')
+    return buildbotClient
+  } catch (error) {
+    addErrorInfo(error, { type: 'buildbotClientStartup' })
+    throw error
+  }
+}
+
+const closeBuildbotClient = async function(buildbotClient) {
+  if (buildbotClient === undefined) {
+    return
+  }
+  await promisify(buildbotClient.end.bind(buildbotClient))()
```

Comment:
Should errors in `closeBuildbotClient` have their own error type like `buildbotClientStartup`? 
Another option would be for both of those to have a common error type for "anything that goes wrong with the TCP connection, but probably should not", e.g. `buildbotClientConnection` error type?

Reply:
I think the second option would work well; I'll do that.